src/Renci.SshNet/Abstractions/SocketAbstraction.cs

```diff
@@ -325,9 +339,16 @@ public static int Read(Socket socket, byte[] buffer, int offset, int size, TimeS
             return totalBytesRead;
         }
 
-        public static Task<int> ReadAsync(Socket socket, byte[] buffer, int offset, int length, CancellationToken cancellationToken)
+        public static Task<int> ReadAsync(Socket socket, byte[] buffer, CancellationToken cancellationToken)
         {
-            return socket.ReceiveAsync(buffer, offset, length, cancellationToken);
+#if FEATURE_SOCKET_TAP
+            Memory<byte> memory = buffer;
+            return socket.ReceiveAsync(memory, SocketFlags.None, cancellationToken).AsTask();
```

Comment:
In this pleace, I think it is better to use the native method `ReceiveAsync` instead of our extension.

Reply:
Agreed. I did not even realise it was using an extension. Maybe wrap the extension in `#if !NET`? Could also inline this method into the (single) call site so that the `AsTask` is not needed.

Reply:
`AsTask()` is required because a new `ReceiveAsync` return `ValueTask`.

Reply:
I meant that this function is only called in `ProtocolVersionExchange` so we could remove it from `SocketAbstraction` and await the `ValueTask` in `ProtocolVersionExchange`. But we can keep it like this